solidity/contracts/isms/hook/ArbL2ToL1Ism.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {Message} from "../../libs/Message.sol";
+import {AbstractMessageIdAuthorizedIsm} from "./AbstractMessageIdAuthorizedIsm.sol";
+
+// ============ External Imports ============
+
+import {IOutbox} from "@arbitrum/nitro-contracts/src/bridge/IOutbox.sol";
+import {CrossChainEnabledArbitrumL1} from "@openzeppelin/contracts/crosschain/arbitrum/CrossChainEnabledArbitrumL1.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title ArbL2ToL1Ism
+ * @notice Uses the native Arbitrum bridge to verify interchain messages from L2 to L1.
+ */
+contract ArbL2ToL1Ism is
+    CrossChainEnabledArbitrumL1,
+    AbstractMessageIdAuthorizedIsm
+{
+    using Message for bytes;
+    // ============ Constants ============
+
+    // module type for the ISM
+    uint8 public constant moduleType =
+        uint8(IInterchainSecurityModule.Types.ARB_L2_TO_L1);
+    // arbitrum nitro contract on L1 to forward verification
+    IOutbox public arbOutbox;
+
+    // ============ Constructor ============
+
+    constructor(
+        address _bridge,
+        address _outbox
+    ) CrossChainEnabledArbitrumL1(_bridge) {
+        require(
+            Address.isContract(_bridge),
+            "ArbL2ToL1Ism: invalid Arbitrum Bridge"
+        );
+        arbOutbox = IOutbox(_outbox);
+    }
+
+    // ============ External Functions ============
+
+    /// @inheritdoc IInterchainSecurityModule
+    function verify(
+        bytes calldata metadata,
+        bytes calldata message
+    ) external override returns (bool) {
+        return
+            _statefulVerify(metadata, message) ||
```

Comment:
why is this `||`?

Reply:
afaict `_statefulVerify` does not verify authorized sender so this could be exploited

Reply:
the first condition is when someone has already called; the second is when we call executeTransaction internally. _statefulVerify doesn't need, verifyMessageId already did it.